modules/deploy.js

```diff
@@ -63,127 +60,60 @@ async function localFilesListing(dir, files2Ignore, firstLevel = false) {
 
 
 function findChanges(files, config, options) {
-  return new Promise((resolve, reject) => {
-
-    let url = cliConfs.getApiUrl() + 'instances/' + config.site_name + "/changes";
-
-    let form = Object.assign({}, { "files": JSON.stringify(files) });
-    form.location_str_id = options.location_str_id;
-
-    request.post({
-      headers: {
-        "x-auth-token": config.token
-      },
-      url: url,
-      json: true,
-      formData: form
-    }, function optionalCallback(err, httpResponse, body) {
-
-      if (err || httpResponse.statusCode != 200) {
-        reject("failed to obtain changes");
-      } else {
-        resolve(body);
-      }
-    });
+  let form = Object.assign({}, {
+    "files": JSON.stringify(files)
   });
+  form.location_str_id = options.location_str_id;
+  return fetch.post('instances/' + config.site_name + "/changes", form, config)
 }
 
-function sendCompressedFile(file, config, options) {
-  return new Promise((resolve, reject) => {
-
-    let formData = {
-      "info": JSON.stringify({"path": file}),
-      "version": config.version,
-      "location_str_id": options.location_str_id
-    };
-
-    let file2Upload = fs.createReadStream(file);
-    formData.file = file2Upload
-
-    let url = cliConfs.getApiUrl() + 'instances/' + config.site_name +
-      "/sendCompressedFile";
-
-    request.post({
-      headers: {
-        "x-auth-token": config.token
-      },
-      url: url,
-      formData: formData
-    }, function optionalCallback(err, httpResponse, body) {
-      if (err || httpResponse.statusCode != 200) {
-        reject(body);
-      } else {
-        resolve(body);
-      }
-    });
-  });
+async function sendCompressedFile(file, config, options) {
+  let formData = new FormData()
+  let file2Upload = fs.createReadStream(file);
+  formData.append('file', file2Upload)
+  formData.append('info', JSON.stringify({
+      "path": file
+    }))
+  formData.append('version', config.version)
+  formData.append('location_str_id', options.location_str_id)
+  const Length = await new Promise((resolve, reject) => {
+    formData.getLength(function(err, length) {
+      resolve(length)
+    })
+  })
+  return fetch.upload('instances/' + config.site_name + "/sendCompressedFile", formData, config, null, Length)
```

Comment:
better to stringify with:

`instances/${config.site_name}/sendCompressedFile`

Reply:
best would be to change it every where.... :(